pandas/tests/series/test_constructors.py

```diff
@@ -303,9 +299,27 @@ def test_constructor_pass_nan_nat(self):
     def test_constructor_cast(self):
```

Comment:
This test name isn't particularly informative.  Let's break this test up so that we can check for specific errors and also utilize `pytest.mark.parametrize` for cleaner tests.
~~~
def test_constructor_unsigned_dtype_overflow(self):
...
def test_constructor_coerce_float_fail(self:
...
~~~
Also, I prefer if we can use `tm.assert_raises_regex` to specify specific error messages instead of just testing for the `Exception` type.  That's a stronger test IMO.  The test above:
~~~python
pytest.raises(ValueError, Series, ['a', 'b', 'c'], dtype=float)
~~~
I can see begin to understand why it fails, but what is the exact reason where it is breaking?  If it is separate from the ones you added, let's make that a test of its own.